File: BackEnd/backend/src/main/java/com/fitapp/backend/application/dto/user/UserResponse.java
package com.fitapp.backend.application.dto.user;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.LocalDateTime;
import com.fitapp.backend.application.dto.subscription.SubscriptionResponse;
import com.fitapp.backend.infrastructure.persistence.entity.enums.Role;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class UserResponse {
    @JsonProperty("id")
    private Long id;
    
    @JsonProperty("createdAt")
    private LocalDateTime createdAt;
    
    @JsonProperty("updatedAt")
    private LocalDateTime updatedAt;
    
    @JsonProperty("email")
    private String email;
    
    @JsonProperty("fullName")
    private String fullName;
    
    @JsonProperty("role")
    private Role role;
    
    @JsonProperty("isActive")
    private boolean isActive;
    
    @JsonProperty("maxRoutines")
    private Integer maxRoutines;
    
    @JsonProperty("subscription")
    private SubscriptionResponse subscription;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/feature/profile/model/response/UserResponse.kt
package com.fitapp.appfit.feature.profile.model.response

import com.fitapp.appfit.feature.subscription.model.response.SubscriptionResponse
import com.google.gson.annotations.SerializedName

data class UserResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("email") val email: String,
    @SerializedName("fullName") val fullName: String?,
    @SerializedName("role") val role: String,
    @SerializedName("isActive") val isActive: Boolean,
    @SerializedName("emailVerified") val emailVerified: Boolean = false,
    @SerializedName("subscription") val subscription: SubscriptionResponse?
)